packages/nova-react/src/eventing/nova-eventing-provider.tsx

```diff
@@ -47,57 +57,92 @@ export interface NovaReactEventing {
   generateEvent(eventWrapper: GeneratedEventWrapper): Promise<void>;
 }
 
-export const NovaEventingProvider: React.FunctionComponent<NovaEventingProviderProps> =
-  ({ children, eventing, reactEventMapper }) => {
-    // Nova contexts provide a facade over framework functions
-    // We don't need to trigger rerender in children when we are rerendered
-    // or when the input functions change, we just need to make sure callbacks
-    // use the right functions
-    const eventingRef = React.useRef(eventing);
-    if (eventingRef.current !== eventing) {
-      eventingRef.current = eventing;
-    }
+export const NovaEventingProvider: React.FunctionComponent<
+  NovaEventingProviderProps
+> = ({ children, eventing, unmountEventing, reactEventMapper }) => {
+  // Nova contexts provide a facade over framework functions
+  // We don't need to trigger rerender in children when we are rerendered
+  // or when the input functions change, we just need to make sure callbacks
+  // use the right functions
+  const eventingRef = React.useRef(eventing);
+  if (eventingRef.current !== eventing) {
+    eventingRef.current = eventing;
+  }
 
-    const mapperRef = React.useRef(reactEventMapper);
-    if (mapperRef.current !== reactEventMapper) {
-      mapperRef.current = reactEventMapper;
-    }
+  const unmountEventingRef = React.useRef(unmountEventing || eventing);
+  if (unmountEventingRef.current !== unmountEventing) {
+    unmountEventingRef.current = unmountEventing || eventing; // default to eventing if unmountEventing not supplied
+  }
 
-    const reactEventing: NovaReactEventing = React.useMemo(
-      () => ({
-        bubble: (eventWrapper: ReactEventWrapper) => {
-          const mappedEvent: EventWrapper = mapperRef.current(eventWrapper);
-          return eventingRef.current.bubble(mappedEvent);
-        },
-        generateEvent: (eventWrapper: GeneratedEventWrapper) => {
-          const mappedEvent = {
-            event: eventWrapper.event,
-            source: {
-              inputType: InputType.programmatic,
-              timeStamp: eventWrapper.timeStampOverride ?? Date.now(),
-            },
-          };
-          return eventingRef.current.bubble(mappedEvent);
-        },
-      }),
-      [],
-    );
+  const mapperRef = React.useRef(reactEventMapper);
+  if (mapperRef.current !== reactEventMapper) {
+    mapperRef.current = reactEventMapper;
+  }
+
+  const reactEventing = React.useMemo(
+    generateEventing(eventingRef, mapperRef),
+    [],
+  );
+
+  const reactUnmountEventing = React.useMemo(
+    generateEventing(unmountEventingRef, mapperRef),
+    [],
+  );
 
-    return (
-      <NovaEventingContext.Provider value={reactEventing}>
-        {children}
-      </NovaEventingContext.Provider>
-    );
-  };
+  return (
+    <NovaEventingContext.Provider
+      value={{ eventing: reactEventing, unmountEventing: reactUnmountEventing }}
```

Comment:
As long as the provider does not re-render, this will not be an issue. However, every time the NovaEventingContext.Provider re-renders, all children will re-render, since the object reference assigned to the value prop is created inline. Please consider stabilizing this object reference with a memo.